### PR TITLE
issue/2363 drop src/course pathway in grunt

### LINKED ISSUE #2363: Drop 'src/course' folder in favour of 'build/course'
### Advantages
* makes src folder "code only"
* unifies build pathways across AAT and Fwk
* deduplicates course assets when stored on disk
* allows developers to concentrate on build output
* simplifies grunt code

### Suggested timeline
Release with v6

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes #2363 `build/course` or supplied `outputdir` folder only, no `src/course`
* Changed to 2 spaces for json output
* Added json-prettify task for dev to compliment json-minify for build
* Reorganised main grunt tasks around `server-build`
* Allowed `build:diff` and `dev:diff` as options during refactor
* Deleted `src/course` folder, removed pathways for processing
* Added `build/course` folder, .gitignore and copy task directives
* Ensure `grunt clean` works correctly

build/course - course content
src - plugins only
src/course - deleted

IMPORTANT:
This will break AAT import and export as they expect src/course.
It will not break the framework but is a sufficiently large change to warrant a major version number.

QUESTION:
~~Process - Do we need a separate branch for v6 (as with v5) or merge straight into master?~~ Yes.

### CODE REVIEW

File: grunt/tasks/build.js [hunk 1]
```diff
@@ -2,21 +2,19 @@
...
-    'less:compile',
-    'replace',
-    'scripts:adaptpostbuild',
-    'minify'
-  ]);
+  grunt.registerTask('build', 'Creates a production-ready build of the course', function(diff) {
+    diff = diff ? ":diff" : '';
+    grunt.task.run([
+      'check-json',
+      'clean:output',
```
Comment (oliverfoster) [inline grunt/tasks/build.js]: Note to self: Turn off clean on a diff

**Comment by brian-learningpool** [comment]
Is there any knock-on effect on the Adapt CLI with this change?

**Comment by oliverfoster** [comment]
> Is there any knock-on effect on the Adapt CLI with this change?

There shouldn't be. I will check.

**Comment by chucklorenz** [comment]
> Do we need a separate branch for v6 (as with v5) or merge straight into master?

Thinking abt those who come new to the project, or who just simply don't take the time to deliberately choose btwn the master and a release: can you avoid some questions/problems/explanations by using a v6 branch? (no strong opinion myself)

**Pull request merged**